### PR TITLE
feat: add schema check on postgres startup

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This patch adds check for `database` parameter to see if it's valid and reject connection accordingly.

It adds `CatalogHandler` to server handler family, and will be open to other apis like http.

This patch also updates pgwire to fix `peek` usage

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/servers/Cargo.toml [hunk 1]
```diff
@@ -34,7 +35,7 @@ num_cpus = "1.13"
 once_cell = "1.16"
 openmetrics-parser = "0.4"
 opensrv-mysql = "0.3"
-pgwire = "0.6.1"
+pgwire = "0.6.2"
```
Comment (MichaelScofield) [inline src/servers/Cargo.toml:38]:
```suggestion
pgwire = "0.6"
```

File: src/servers/src/error.rs [hunk 2]
```diff
@@ -233,6 +234,9 @@ pub enum Error {
         source: FromUtf8Error,
         backtrace: Backtrace,
     },
+
+    #[snafu(display("Error accessing catalog: {}", source))]
+    CatalogError { source: catalog::error::Error },
```
Comment (MichaelScofield) [inline src/servers/src/error.rs:239]: better to include "which" catalog is failed to access

File: src/servers/Cargo.toml (hunk 1 above)
Comment (sunng87) [reply src/servers/Cargo.toml:38]: here we require a minimum version 0.6.2 to trigger an update on Cargo.lock because previous versions have known bug.

**Review by MichaelScofield — APPROVED** [review]

**Review by shuiyisong — APPROVED** [review]
LGTM

**Pull request merged**